File: test/com/barrybecker4/common/profile/MemoryLogger.java
// Copyright by Barry G. Becker, 2013. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.common.profile;

import com.barrybecker4.common.app.ILog;

import java.io.FileNotFoundException;

/**
 * @author Barry Becker
 */
public class MemoryLogger implements ILog {

    StringBuilder bldr;

    @Override
    public void setDestination(int logDestination) {
    }

    @Override
    public int getDestination() {
        return 0;
    }

    @Override
    public void setLogFile(String fileName) throws FileNotFoundException {
    }

    @Override
    public void setStringBuilder(StringBuilder bldr) {
       this.bldr = bldr;
    }

    @Override
    public void print(int logLevel, int appLogLevel, String message) {
    }

    @Override
    public void println(int logLevel, int appLogLevel, String message) {
    }

    @Override
    public void print(String message) {
        bldr.append(message);
    }

    @Override
    public void println(String message) {
        bldr.append(message).append("\n");
    }

    public String toString() {
        return bldr.toString();
    }
}

File: scala-test/com/barrybecker4/common/app/MemoryLogger.scala
/* Copyright by Barry G. Becker, 2000-2018. Licensed under MIT License: http://www.opensource.org/licenses/MIT */
package com.barrybecker4.common.app

import java.io.FileNotFoundException


/**
  * Log to memory only. Mostly used in unit tests, but could have other uses.
  * @author Barry Becker
  */
class MemoryLogger(sbuilder: Option[StringBuilder] = None) extends ILog {
  private var bldr: StringBuilder = if (sbuilder.isDefined) sbuilder.get else null

  override def setDestination(logDestination: Int): Unit = {}
  override def getDestination = 0

  @throws[FileNotFoundException]
  override def setLogFile(fileName: String): Unit = {}
  override def setStringBuilder(bldr: StringBuilder): Unit = {this.bldr = bldr}
  override def print(logLevel: Int, appLogLevel: Int, message: String): Unit = {}
  override def println(logLevel: Int, appLogLevel: Int, message: String): Unit = {}
  override def print(message: String): Unit = { bldr.append(message) }
  override def println(message: String): Unit = { bldr.append(message).append("\n")}
  override def toString: String = bldr.toString
}
